libsolidity/codegen/ContractCompiler.cpp

```diff
@@ -426,6 +410,13 @@ bool ContractCompiler::visit(FunctionDefinition const& _function)
 
 	m_context.startFunction(_function);
 
+	// Reject transaction if value is not accepted, but was received
+	if (!_function.isPayable())
```

Comment:
Question is, should it be in `visit(FunctionDefiniton)` or `appendFunctionSelector`?